File: application/src/test/java/com/megustav/lolesports/schedule/processor/TestProcessorRepository.java
package com.megustav.lolesports.schedule.processor;

import com.megustav.lolesports.schedule.configuration.ProcessorConfiguration;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import java.util.Optional;
import java.util.stream.Stream;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * Processors testing
 *
 * @author MeGustav
 *         06/04/2018 17:11
 */
@RunWith(SpringRunner.class)
@SpringBootTest(classes = ProcessorConfiguration.class)
public class TestProcessorRepository {

    /** Processor repository */
    @Autowired
    private ProcessorRepository repository;

    /**
     * Test whether all processors were registered
     */
    @Test
    public void testAllProcessorsRegistered() {
        Stream.of(ProcessorType.values())
                .forEach(type -> {
                    Optional<MessageProcessor> processor = repository.getProcessor(type);
                    assertThat(processor.isPresent())
                            .as("Processor of type '" + type + "' registered")
                            .isTrue();
                });
    }
}

File: application/src/test/kotlin/com/megustav/lolesports/schedule/context/TestProcessorRepository.kt
package com.megustav.lolesports.schedule.context

import com.megustav.lolesports.schedule.configuration.ProcessorConfiguration
import com.megustav.lolesports.schedule.processor.ProcessorRepository
import com.megustav.lolesports.schedule.processor.ProcessorType
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.annotation.DirtiesContext
import org.springframework.test.context.junit4.SpringRunner
import kotlin.test.assertTrue

/**
 * @author MeGustav
 *         2019-02-14 20:12
 */
@RunWith(SpringRunner::class)
@SpringBootTest(classes = [ProcessorConfiguration::class])
@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_CLASS)
class TestProcessorRepository {

    @Autowired
    private lateinit var repository: ProcessorRepository

    /**
     * Test whether all processors were registered
     */
    @Test
    fun testAllProcessorsRegistered() {

        ProcessorType.values().forEach {
            assertTrue { repository.getProcessor(it).isPresent }
        }
    }

}